src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -127,6 +127,34 @@ rpmostreed_deployment_generate_blank_variant (void)
 }
 
 static void
+variant_add_metadata_attribute (GVariantDict *dict,
+                                const gchar  *attribute,
+                                GVariant     *commit)
+{
+  g_autoptr(GVariant) metadata = NULL;
+  g_autofree gchar *attribute_string_value = NULL;
+
+  metadata = g_variant_get_child_value (commit, 0);
+
+  if (metadata != NULL)
+    {
+      g_variant_lookup (metadata, attribute, "s", &attribute_string_value);
+      if (attribute_string_value != NULL)
+        {
+          /* for attributes starting with ostree., remove the prefix */
```

Comment:
This seems kinda arbitrary to generalize given that it's only used for one field right now. I'd just make `attribute` and `new_attribute` parameters, though it's up to you!